lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -281,6 +298,36 @@ newDBFactory tr defaultFieldValues mDatabaseDir = do
                 databaseFilePrefix <> "." <>
                 T.unpack (toText wid) <> ".sqlite"
 
+internalRemovePathForcibly :: FilePath -> IO ()
+internalRemovePathForcibly =
+#if defined(mingw32_HOST_OS)
+    -- On Windows we have to retry the delete a couple of times.
+    -- The reason for this is that a FileDelete command just marks a
+    -- file for deletion. The file is really only removed when the last
+    -- handle to the file is closed. Unfortunately there are a lot of
+    -- system services that can have a file temporarily opened using a shared
+    -- read-only lock, such as the built in AV and search indexer.
+    --
+    -- We can't really guarantee that these are all off, so what we can do is
+    -- whenever after an rm the file still exists to try again and wait a bit.
+    retryOnPermissionErrors 5 . removePathForcibly
+#else
+    removePathForcibly
+#endif
```

Comment:
We'd need to check if there's a way to check / make sure that the SQLite has been properly closed before removing the file. In practice, we always:

a. Kill the thread that has the connection open (which, because of bracketting-style resource allocation should free-up the connection at some point)

b. Only remove the database file when we know that the connection has ended. 

On Linux, it seems that either the connection is released fast enough or, that it doesn't matter if the connection is still open when the file is deleted. On Windows, that's a different story indeed. 

One possible "easy way" to work around this I see would be to have an MVar shared between the `withDatabase` and `removeDatabase` such that, `removeDatabase` would only attempt to remove the file once the connection has been terminated on `withDatabase`; so waiting on an MVar before `mapM_ removePathForcibly ...` with a protective time-out. 

cc @rvl

Reply:
@jbgi This is interesting. What references do you have for these claims?

1. "On Windows ... a FileDelete command just marks a file for deletion. The file is really only removed when the last handle to the file is closed."
2. "there are a lot of system services that can have a file temporarily opened using a shared read-only lock, such as the built in AV and search indexer."

For 1, I thought that was more like the behaviour on Linux (although on Linux, the directory entry is immediately removed, but the inode stays). Do you mean that the filename lingers in the directory listing untll it has actually been deleted?

For 2, do these services _really_ cause files to not be deleted promptly? I thought that would break lots of things that expect FileDelete to do the obvious.

Reply:
main reference: https://github.com/haskell/directory/issues/96
I originally implemented this "fix" in order to try getting a test pass on wine, but in the end, it was not necessary for the test to pass, or at least not enough, though I still think this hack is useful from what I have read.